src/Compilers/CSharp/Portable/Symbols/Source/SourceDelegateClonedParameterSymbolForBeginAndEndInvoke.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SourceDelegateClonedParameterSymbolForBeginAndEndInvoke : SourceClonedParameterSymbol
+    {
+        private const int UninitializedArgumentExpressionParameterIndex = int.MinValue;
+        private int _lazyCallerArgumentExpressionParameterIndex = UninitializedArgumentExpressionParameterIndex;
```

Comment:
>private int _lazyCallerArgumentExpressionParameterIndex = UninitializedArgumentExpressionParameterIndex; [](http://example.com/codeflow?start=8&length=104)

I am curios, is it worth all the trouble of recalculating the position? Could we just return -1 for now and see if anyone would ask us to support the scenarios.